src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -94,24 +95,50 @@ protected override DefinitionRegistrationOptions CreateRegistrationOptions(Defin
-            {
-                // goto beginning of the module file.
-                return Task.FromResult(GetModuleDefinitionLocation(
-                    GetDocumentLinkUri(sourceFile, moduleReference),
-                    stringToken,
-                    context,
-                    new Range { Start = new Position(0, 0), End = new Position(0, 0) }));
+            { // Definition handler for a non symbol bound to implement module path goto.
+                // try to resolve module path syntax from given offset using tail matching.
+                if (SyntaxMatcher.IsTailMatch<ModuleDeclarationSyntax, StringSyntax, Token>(
+                     matchingNodes,
+                     (moduleSyntax, stringSyntax, token) => moduleSyntax.Path == stringSyntax && token.Type == TokenType.StringComplete)
+                 && matchingNodes[^3] is ModuleDeclarationSyntax moduleDeclarationSyntax
+                 && matchingNodes[^2] is StringSyntax stringToken
+                 && context.Compilation.SourceFileGrouping.TryLookUpModuleSourceFile(moduleDeclarationSyntax) is ISourceFile sourceFile
+                 && this.moduleDispatcher.TryGetModuleReference(moduleDeclarationSyntax, context.Compilation.Configuration, out _) is { } moduleReference)
+                {
+                    // goto beginning of the module file.
+                    return Task.FromResult(GetFileDefinitionLocation(
+                        GetDocumentLinkUri(sourceFile, moduleReference),
+                        stringToken,
+                        context,
+                        new Range { Start = new Position(0, 0), End = new Position(0, 0) }));
+                }
+            }
+            {  // Definition handler for a non symbol bound to implement load* functions file argument path goto.
+                if (SyntaxMatcher.IsTailMatch<FunctionCallSyntaxBase, FunctionArgumentSyntax, StringSyntax, Token>(
+                        matchingNodes, (_, functionArgumentSyntax, stringSyntax, token) =>
+                            functionArgumentSyntax.Expression == stringSyntax
+                            && !stringSyntax.IsInterpolated()
+                            && token.Type == TokenType.StringComplete)
+                    && matchingNodes[^2] is StringSyntax stringToken
+                    && matchingNodes[^3] is FunctionArgumentSyntax functionArgument
+                    && matchingNodes[^4] is FunctionCallSyntaxBase functionCall
+                    && stringToken.TryGetLiteralValue() is { } stringTokenValue
+                    && !string.IsNullOrWhiteSpace(stringTokenValue)
+                    && context.Compilation.GetEntrypointSemanticModel().GetSymbolInfo(functionCall) is FunctionSymbol functionSymbol
+                    && functionSymbol.GetDeclaredArgumentType(functionCall.Arguments.ToList().IndexOf(functionArgument)) is { } argumentType
+                    && argumentType.ValidationFlags.HasFlag(TypeSymbolValidationFlags.IsStringFilePath)
```

Comment:
Can this logic now be simplified now that you're using validation flags? For example, something like:
```csharp
if (SyntaxMatcher.IsTailMatch<StringSyntax, Token>(_, token) => token.Type == TokenType.StringComplete) &&
    context.Compilation.GetEntrypointSemanticModel().GetDeclaredType(matchingNodes[^2]) is {} stringType &&
    stringType.ValidationFlags.HasFlag(TypeSymbolValidationFlags.IsStringFilePath) &&
    stringToken.TryGetLiteralValue() is { } stringTokenValue &&
    fileResolver.TryResolveFilePath(context.Compilation.SourceFileGrouping.EntryPoint.FileUri, stringTokenValue) is { } fileUri &&
    fileResolver.FileExists(fileUri))
```